plugins/sm-operator/sm-operator/__init__.py

```diff
@@ -0,0 +1,105 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License").
+#    You may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import logging
+import os
+from typing import TYPE_CHECKING, Any, Dict, Optional, cast
+
+from aws_orbit import sh, utils
+from aws_orbit.plugins import hooks
+from aws_orbit.remote_files import helm
+
+
+if TYPE_CHECKING:
+    from aws_orbit.models.context import Context, TeamContext
+_logger: logging.Logger = logging.getLogger("aws_orbit")
+CHART_PATHS = os.path.join(os.path.dirname(__file__))
+
+@hooks.deploy
+def deploy(plugin_id: str, context: "Context", team_context: "TeamContext", parameters: Dict[str, Any]) -> None:
+    _logger.debug("Team Env name: %s | Team name: %s", context.name, team_context.name)
+    plugin_id = plugin_id.replace("_", "-")
+    _logger.debug("plugin_id: %s", plugin_id)
+    release_name = f"{team_context.name}-{plugin_id}"
+
+    ### Logic
```

Comment:
I think this logic is flawed when we deal with multiple teams.  What will happen if the CRD was installed by another team.  The CRD needs to be installed once and not per team and we should separate it out of the team package.   I recommend moving the CRD into 0-commons of the env under cli/aws_orbit/data/kubectl/env/00-commons.yaml

Reply:
Now I actually think this is bigger than this comment.  We need to create two different helm packages. One cluster level and second will be team level.  We can check if the cluster level is deployed and only update it then, and install the team level .    All the cluster level resources should go there {ClusterRole, ClusterRoleBinding(need to fix it to have unique name per team),  CRD (which I cannot find in this PR) }

Reply:
I also need to understand if we are sure that we need to deploy this service once per team or once per cluster.  It looks like one per team now, so how does the notebook in each team space talks to the right controller? Lets verify this first.

Reply:
CRD's located here: plugins/sm-operator/sm-operator/crds/crd.yaml

I believe the service needs to be installed with a namespace to promote the isolation that Orbit promotes...?   

The notebook I provided shows an example:  current_job = !kubectl apply -f {job_yaml} -n lake-user
(though the lake user should be parameterized).

Let's have a design session?  If we put all CRD's into a separate helm package and call it when creating the env, what are the implications...?